src/main/java/ladysnake/satin/mixin/client/render/WorldRendererMixin.java

```diff
@@ -0,0 +1,44 @@
+package ladysnake.satin.mixin.client.render;
+
+import ladysnake.satin.impl.BlockRenderLayerRegistryImplementation;
+import net.minecraft.client.render.RenderLayer;
+import net.minecraft.client.render.WorldRenderer;
+import net.minecraft.client.util.math.MatrixStack;
+import net.minecraft.resource.SynchronousResourceReloadListener;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Redirect;
+import org.spongepowered.asm.mixin.injection.Slice;
+
+@Mixin(WorldRenderer.class)
+public abstract class WorldRendererMixin implements SynchronousResourceReloadListener, AutoCloseable{
+    @Shadow protected abstract void renderLayer(RenderLayer layer, MatrixStack matrix, double x, double y, double z);
+    
+    @Redirect(
+        method = "render",
+        at = @At(
+            value = "INVOKE",
+            target = "Lnet/minecraft/client/render/WorldRenderer;renderLayer(Lnet/minecraft/client/render/RenderLayer;Lnet/minecraft/client/util/math/MatrixStack;DDD)V",
+            ordinal = 2
+        ),
```

Comment:
You do not need a redirect, you can just capture the variables as they are computed very early in the method.